### PR TITLE
refactor: change package file structure

### PR DESCRIPTION
**Pull request by Leptopoda**
I didn't know what to do with the cookie_store packages.
Should I move them to the framework, or just keep them as they are?

### CODE REVIEW

**Comment by provokateurin** [comment]
What about moving the `packages/neon/neon_*` packages to `packages/neon_framework/packages` and `packages/dynamite/dynamite_*` to `packages/dynamite/packages`?

**Comment by Leptopoda** [comment]
I agree with dynamite, but am unsure about the neon_* ones.
With my proposed way, each `$consumer/packages` contains the packages that are needed for `$consumer` while moving them into the framework would switch that up.

**Comment by provokateurin** [comment]
I see, they are a bit weird.

BTW you also need to fix all the paths in tool/ and any documentation.

**Comment by provokateurin** [comment]
Maybe cookie_store should go into neon_http_client? OTOH it is a pretty standalone package that we could publish at some point so that others can use it too.

**Comment by Leptopoda** [comment]
> BTW you also need to fix all the paths in tool/ and any documentation.

I've tried my best to search for all packages I changed and hope I didn't miss any.

> Maybe cookie_store should go into neon_http_client? 

Let's leave it there for now.

**Comment by provokateurin** [comment]
Instead of making a long and tedious review I just fixed all the problems. I found quite a few, but I think now everything is fixed.

**Comment by provokateurin** [comment]
@Leptopoda Let me know if you find my changes good, then I'll approve and we can merge :rocket:

**Comment by provokateurin** [comment]
BTW I already love working with this structure, it is so much better than what we had before! :heart:

File: packages/neon_framework/example/pubspec.lock [hunk 1]
```diff
@@ -1666,5 +1666,5 @@ packages:
     source: hosted
     version: "3.1.2"
 sdks:
-  dart: ">=3.4.0 <4.0.0"
+  dart: ">=3.5.0 <4.0.0"
```
Comment (provokateurin) [inline packages/neon_framework/example/pubspec.lock:1669]: I think this might cause the analyzer issues? At least it looks weird to me because there should be nothing that is causing this :thinking:

Comment (provokateurin) [reply packages/neon_framework/example/pubspec.lock]: I was able to revert the unintended changes to the pubspec.lock, but it didn't fix the problem.

Comment (Leptopoda) [reply packages/neon_framework/example/pubspec.lock]: I can confirm your findings.
Pinning all sdk versions to 3.4 does not trigger this issue.
The `context.mounted` not working is still an issue that requires us to ignore the errors (I'll report upstream).

File: .github/renovate.json [hunk 2]
```diff
@@ -30,7 +30,6 @@
       ],
       "matchCurrentVersion": "!/^0/",
       "matchDepNames": [
-        "!app",
```
Comment (Leptopoda) [inline .github/renovate.json:33]: Why this change?

Comment (provokateurin) [reply .github/renovate.json:33]: Because the example packages shouldn't be treated like internal packages where we always update to the latest version.

Comment (Leptopoda) [reply .github/renovate.json:33]: but shouldn't we always use the latest version for all our examples?
1. verify that our examples still work with the latest versions.
2. with this being the app we ship to end users, I'd like to make sure to pull in all available fixes

Comment (provokateurin) [reply .github/renovate.json:33]: Hm ok makes sense

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
:tada:
